fedora_messaging/config.py

```diff
@@ -393,9 +356,56 @@ def setup_logging(self):
             self.load_config()
         logging.config.dictConfig(self['log_config'])
 
-    def load_config(self, filename=None):
+    def _validate(self):
+        """
+        Perform checks on the configuration to assert its validity
+
+        Raises:
+            ConfigurationException: If the configuration is invalid.
+        """
+        for key in self:
+            if key not in DEFAULTS:
+                raise exceptions.ConfigurationException(
+                    'Unknown configuration key "{}"! Valid configuration keys are'
+                    ' {}'.format(key, list(DEFAULTS.keys())))
+
+        # Don't let users override these
+        self['client_properties']['version'] = DEFAULTS['client_properties']['version']
+        self['client_properties']['information'] = DEFAULTS['client_properties']['information']
+        self['client_properties']['product'] = DEFAULTS['client_properties']['product']
```

Comment:
It might be good to check if these are overridden and raise an exception. This way users will know that their overridden settings aren't accepted.